integrations/pgvector/src/haystack_integrations/components/retrievers/pgvector/embedding_retriever.py

```diff
@@ -139,7 +144,11 @@ def run(
 
         :returns: List of Documents similar to `query_embedding`.
         """
-        filters = filters or self.filters
+        if self.filter_policy == "merge" and filters:
```

Comment:
In the docstrings of the `run` method, I would mention that the way runtime filters are applied depends on the `filter_policy` chosen at document store initialization.